src/components/productCard/ProductCard.tsx

```diff
@@ -277,7 +273,11 @@ export default function ProductCard({
             });
           }
         }}
-        price={Number(displayFloat(price?.price || 0))}
+        price={
+          /e/i.test(String(Number(price?.price || 0)))
+            ? Number(price?.price || 0).toFixed(10)
+            : (price?.price ?? 0).toString()
+        }
```

Comment:
I think it should be like this
```suggestion
        price={price?.price}
```